api/models/templates/app.tmpl

```diff
@@ -629,6 +630,132 @@
   {{ end }}
 {{ end }}
 
+{{ define "agent-resources" }}
+  {{ if .App.Agents .Manifest }}
+    "AgentRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Principal": { "Service": [ "lambda.amazonaws.com" ] },
+              "Action": [ "sts:AssumeRole" ]
+            }
+          ]
+        },
+        "Path": "/convox/",
+        "Policies": [
+          {
+            "PolicyName": "AgentJob",
+            "PolicyDocument": {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Effect": "Allow",
+                  "Action": "ecs:*",
+                  "Resource": "*",
+                  "Condition": {
+                    "ArnEquals": {
+                      "ecs:cluster": { "Fn::Join": [ "", [
+                        "arn:aws:ecs:",
+                        { "Ref": "AWS::Region" },
+                        ":",
+                        { "Ref": "AWS::AccountId" },
+                        ":cluster/",
+                        "{{ env "CLUSTER" }}"
+                      ] ] }
+                    }
+                  }
+                },
+                {
+                  "Effect": "Allow",
+                  "Action": [
+                    "logs:CreateLogGroup",
+                    "logs:CreateLogStream",
+                    "logs:PutLogEvents"
+                  ],
+                  "Resource": { "Fn::Join": [ "", [ "arn:aws:logs:", { "Ref": "AWS::Region" }, ":", { "Ref": "AWS::AccountId" }, ":log-group:/aws/lambda/*" ] ] }
+                },
+                {
+                  "Effect": "Allow",
+                  "Action": [
+                    "logs:CreateLogStream",
+                    "logs:PutLogEvents"
+                  ],
+                  "Resource": { "Fn::Join": [ "", [ { "Fn::GetAtt": [ "LogGroup", "Arn" ] }, ":*" ] ] }
+                }
+              ]
+            }
+          }
+        ]
+      }
+    },
+    "AgentFunction": {
+      "Type": "AWS::Lambda::Function",
+      "DependsOn": [ "AgentRole" ],
+      "Properties": {
+        "Handler": "index.handler",
+        "Role": { "Fn::GetAtt": [ "AgentRole", "Arn" ] },
+        "Runtime": "nodejs4.3",
+        "Timeout": 50,
+        "Code": {
+          "ZipFile": { "Fn::Join": ["\n", [
+            {{ $code := .App.AgentFunctionCode }}
+            {{ index $code "head" }}
+            {{ range .App.Agents .Manifest }}
```

Comment:
I could move more code below the `{{ range }}`, but you'd still end up with

```
'use strict';
const taskDefinitions = {
```

either written directly in the template, or returned by `AgentFunctionCode`.